packages/components/src/toggle-control/index.js

```diff
@@ -7,62 +7,48 @@ import classnames from 'classnames';
 /**
  * WordPress dependencies
  */
-import { Component } from '@wordpress/element';
-import { withInstanceId } from '@wordpress/compose';
+import { useInstanceId } from '@wordpress/compose';
 
 /**
  * Internal dependencies
  */
 import FormToggle from '../form-toggle';
 import BaseControl from '../base-control';
 
-class ToggleControl extends Component {
-	constructor() {
-		super( ...arguments );
-
-		this.onChange = this.onChange.bind( this );
+export default function ToggleControl( {
+	label,
+	checked,
+	help,
+	className,
+	onChange,
+} ) {
+	function onChangeToggle( event ) {
+		onChange( event.target.checked );
```

Comment:
The previous onChange handler was checking whether the onChange prop was set or not. Should we do the same?

Reply:
The onChange prop is required, without this prop the component will not be usable. When prop is undefined the JS engine will report an error, so the programmer will know what he did wrong.

Reply:
Makes sense and it's also documented as such in the README.